model_analyzer/record/record.py

```diff
@@ -23,15 +23,48 @@
 # OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-import time
 
+from abc import ABC, abstractmethod
 
-class Record:
+
+class Record(ABC):
     """
     This class is used for representing monitoring
     records
     """
+    @abstractmethod
+    def header(self):
+        """
+        Returns
+        -------
+        str
+            The full name of the
+            metric.
+        """
+
+    @abstractmethod
+    def value(self):
+        """
+        This method returns
+        the value of recorded
+        metric
+
+        Returns
+        -------
+        float
+            value of the metric
+        """
+
+    @abstractmethod
+    def timestamp(self):
```

Comment:
This functionality is common to all inherited classes, and you are duplicating the work in all derived classes. Instead, move the actual time variable into this class and then set that time in an _init method for this parent class. I assume inheritence works the same in Python, so the init of a parent class is called. The only thing that should go in child classes are code that is unique to that class. In this case, that is just the header. All other code can be in the parent class